spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/serviceregistry/EurekaAutoServiceRegistration.java

```diff
@@ -52,41 +54,44 @@
 
 	private EurekaServiceRegistry serviceRegistry;
 
-	private EurekaRegistration registration;
+	private ObjectProvider<EurekaRegistration> registrationProvider;
```

Comment:
Did you try without object provider? Refresh scope creates a proxy which should handle getting the refreshed instance for you.

Reply:
Khmm, will try, was thinking only about lazy init as proxy, but not refreshed bean

Reply:
works, fine, reverted class change